### ISSUE TITLE
Assertion `auth_file_fd >= 0' failed

### ISSUE THREAD

**Issue by retifrav** [vysheng/tg#513]
So, I'm getting this error after I type the code from SMS.

<pre>
root@localhost:~/tg# bin/telegram-cli -k ./tg-server.pub -vvvv
Telegram-cli version 1.2.0, Copyright (C) 2013-2015 Vitaly Valtman
Telegram-cli comes with ABSOLUTELY NO WARRANTY; for details type `show_license'.
This is free software, and you are welcome to redistribute it
under certain conditions; type `show_license' for details.
Telegram-cli uses libtgl version 1.2.0
I: config dir=[/home/telegramd/.telegram-cli]
 *** public key './tg-server.pub' loaded successfully
 *** public key '/etc/telegram-cli/server.pub' loaded successfully
 *** public key 'tg-server.pub' loaded successfully
 *** DC1 '' update: 149.154.175.50:443
 *** DC2 '' update: 149.154.167.51:443
 *** DC3 '' update: 174.140.142.6:443
 *** DC4 '' update: 149.154.167.91:443
 *** DC5 '' update: 149.154.171.5:443
 *** outbound rpc connection from dc #4 becomed ready
 *** outbound rpc connection from dc #2 becomed ready
 *** outbound rpc connection from dc #1 becomed ready
 *** outbound rpc connection from dc #5 becomed ready
 *** adjusting CLOCK_MONOTONIC delta to 0.221383
 *** DC3 '' update: 149.154.175.100:443
 *** adjusting CLOCK_MONOTONIC delta to -0.849498
 *** adjusting CLOCK_MONOTONIC delta to 0.280810
 *** adjusting CLOCK_MONOTONIC delta to 0.417028
 *** fail connection: reason: ping timeout
 *** Lost connection to server... 174.140.142.6:80
 *** outbound rpc connection from dc #3 becomed ready
 *** adjusting CLOCK_MONOTONIC delta to -0.760427
phone number: +79876543211
 *** error for query #someid: #303 PHONE_MIGRATE_2 (HANDLED)
 *** Alarm query someid
code ('call' for phone call): code
User myname updated username
User myname offline (was online [2015/05/09 11:57:08])
User myname online (was online [2015/05/09 12:02:29])
</pre>
**telegram-cli: loop.c:360: write_auth_file: Assertion `auth_file_fd >= 0' failed.**
<pre>
SIGNAL received
hbin/telegram-cli(print_backtrace+0x20)[0x4437d0]
bin/telegram-cli(termination_signal_handler+0x64)[0x443854]
/lib/x86_64-linux-gnu/libc.so.6(+0x36d40)[0x7f56f488cd40]
/lib/x86_64-linux-gnu/libc.so.6(gsignal+0x39)[0x7f56f488ccc9]
/lib/x86_64-linux-gnu/libc.so.6(abort+0x148)[0x7f56f48900d8]
/lib/x86_64-linux-gnu/libc.so.6(+0x2fb86)[0x7f56f4885b86]
/lib/x86_64-linux-gnu/libc.so.6(+0x2fc32)[0x7f56f4885c32]
bin/telegram-cli(write_auth_file+0x102)[0x444b42]
bin/telegram-cli(tgl_export_auth_callback+0x66)[0x465716]
bin/telegram-cli[0x45923e]
bin/telegram-cli(tglq_query_result+0xec)[0x45ebcc]
bin/telegram-cli[0x457572]
bin/telegram-cli[0x48129b]
/usr/lib/x86_64-linux-gnu/libevent-2.0.so.5(event_base_loop+0x754)[0x7f56f589ff24]
bin/telegram-cli(net_loop+0xa4)[0x444914]
bin/telegram-cli(loop+0x16e)[0x445c0e]
bin/telegram-cli(main+0x286)[0x442306]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf5)[0x7f56f4877ec5]
bin/telegram-cli[0x4423ee]
</pre>

What am I doing wrong?

**Comment by vysheng** [comment]
telegram-cli by default stores data in ~/.telegram-cli/...
Probably disk is full or you don't have permissions to write to this folder

**Comment by retifrav** [comment]
That's how I started, yes. I'm working with server under root via SSH, and stored that folder in the `/root/`
But when I started `bin/telegram-cli`, it said: `open[config_file]: No such file or directory`

So I created `/home/telegramd/.telegram-cli` and then it started successfully, but return the `auth_file_fd >= 0` error after entering SMS-code.

Now I copied `.telegram-cli` to the `/root/` directory, but the error still remains.

I guees, I messed up with users somewhere?
